### PR TITLE
Provide error message when rest request path is null

### LINKED ISSUE #21232: Avoid NPE in RestClient#buildUri()
**Elasticsearch version**:5.0.0

**Plugins installed**: [none]

**JVM version**:openjdk version "1.8.0_91"

**OS version**:Linux Mint 18

**Description of the problem including expected versus actual behavior**:

1. If we perform a request with null path in RestClient we may got NPE and no hint message.
Since the pathPrefix should not be null, maybe we can treat null path as empty and continue to request.
2. By the way, I have a little question here. Should we encode the URI which composed of pathPrefix and path?

**Steps to reproduce**:
 1.Construct arbitrary valid RestClient.
 2.Perform request with null path.
```restClient.performRequest(method, null);```

**Exception Caught**:
```
java.lang.NullPointerException
	at __randomizedtesting.SeedInfo.seed([D6503272CE20A134:63A320640CEEA8AF]:0)
	at java.net.URI$Parser.parse(URI.java:3042)
	at java.net.URI.<init>(URI.java:588)
	at org.apache.http.client.utils.URIBuilder.<init>(URIBuilder.java:82)
	at org.elasticsearch.client.RestClient.buildUri(RestClient.java:525)
	at org.elasticsearch.client.RestClient.performRequestAsync(RestClient.java:285)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:211)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:185)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:146)
......
```

### PR DESCRIPTION
**Pull request by biyuhao**
Closes #21232

### CODE REVIEW

File: client/rest/src/test/java/org/elasticsearch/client/RestClientIntegTests.java [hunk 1]
```diff
@@ -229,6 +229,18 @@ public void testPathPrefix() throws IOException {
         }
     }
 
+    public void testPath() throws IOException {
+        Response esResponse;
+        for (final String method : getHttpMethods()) {
+            try {
+                esResponse = restClient.performRequest(method, null);
```
Comment (javanna) [inline client/rest/src/test/java/org/elasticsearch/client/RestClientIntegTests.java]: this is not the intended way to use this method. Why should `null` be accepted? I would rather throw NPE earlier than accept it and convert it to something else.

**Review by javanna — CHANGES_REQUESTED** [review]
Thanks for your PR! I left a comment. I think we should improve how we fail, but I don't think we should add leniency and accept null.

File: client/rest/src/test/java/org/elasticsearch/client/RestClientIntegTests.java (hunk 1 above)
Comment (biyuhao) [inline client/rest/src/test/java/org/elasticsearch/client/RestClientIntegTests.java]: Thanks for your review.
Would it be better if we throw a NPE with some message like we do for pathPrefix?
In RestClientBuilder#setPathPrefix() we have:
```Objects.requireNonNull(pathPrefix, "pathPrefix must not be null");```

Comment (javanna) [inline client/rest/src/test/java/org/elasticsearch/client/RestClientIntegTests.java]: yes that would be perfect thanks

Comment (biyuhao) [inline client/rest/src/test/java/org/elasticsearch/client/RestClientIntegTests.java]: I would update this PR soon, thanks again.

**Review by javanna — APPROVED** [review]
thanks a lot @biyuhao LGTM I will merge it soon

**Pull request merged**